website_product_subscription/controllers/subscribe.py

```diff
@@ -15,17 +15,54 @@
 
 
 class SubscribeController(http.Controller):
+    @http.route(
+        "/new/subscription/generic", type="http", auth="public", website=True
+    )
+    def new_subscription_generic(self, **kwargs):
+        request.session["redirect_payment"] = kwargs.get("redirect", "")
+        self.new_subscription_generic_form_validation()
+        if (
+            "error" not in request.params
+            and request.httprequest.method == "POST"
+        ):
+            is_gift = request.params.get("is_gift", False)
+
+            if is_gift:
+                sub_req = self.process_new_subscription_gift_form()
+            else:
+                sub_req = self.process_new_subscription_basic_form()
+
+            values = {
+                "subscription_request_id": sub_req,
+                "subscriber": sub_req.subscriber.id,
+                "subscription_template": sub_req.subscription_template.id,
+                "gift": is_gift,
+                "sponsor": sub_req.sponsor.id if sub_req.sponsor else "",
+            }
+            # Template to render thanks
+            kwargs[
+                "view_callback"
+            ] = "website_product_subscription.product_subscription_thanks"
+            kwargs[
+                "view_callback"
+            ] = "website_product_subscription.product_subscription_thanks"
+            return self.get_subscription_response(values, kwargs)
+
+        return request.website.render(
+            "website_product_subscription.new_subscription_generic", request.params
+        )
+
     @http.route(
         "/new/subscription/basic", type="http", auth="public", website=True
     )
-    def subscribe(self, **kwargs):
+    def new_subscription_basic(self, **kwargs):
         request.session["redirect_payment"] = kwargs.get("redirect", "")
```

Comment:
why using new in all the function names ? it's quite useless no?

Reply:
Yes. If you insist once more, I'll change the route name to `subscription/basic` in a further commit.

Reply:
I insist

Reply:
I can live without the new in the route, even if I found that more explicit to have the verb in the route as here we are not working with a REST API that use verb (PUT, POST, GET,…) to explicit the action.

For the function name, I found it more explicit to have the `new` in it. The previous name was `subscribe` witch is a action verb. What if, in the future, we need to create a route and a function to edit a particular subscription ? It could be `edit_subscription_basic()`.

Although, you will never use this method name except for overwriting it.

That's my opinion, but I can live without the `new` in the route and in the function name.